src/main/gui/main-window.ts

```diff
@@ -399,7 +405,82 @@ export const createMainWindow = async (args: OpenArguments, settings: ChainnerSe
     });
 
     const progressController = new ProgressController();
-    addSplashScreen(progressController);
+
+    let progressFinished = false;
```

Comment:
Could you please move all of this added code (so from `let progressFinished` to `progressController.addInterruptListener`) into its own function? `createMainWindow` is already quite complex.